src/main/scala/uncore/tilelink2/RationalCrossing.scala

```diff
@@ -0,0 +1,177 @@
+// See LICENSE.SiFive for license details.
+
+package uncore.tilelink2
+
+import Chisel._
+import chisel3.internal.sourceinfo.SourceInfo
+import config._
+import diplomacy._
+import util._
+
```

Comment:
A few random ideas for names:

- EvenlyDivisible
- Divisible
- Multiple
- Congruent (because a \congruent b mod min(a, b), though a = k*b is a stronger assertion than saying they have the same remainder)